link-check.js

```diff
@@ -23,37 +33,48 @@ function allMatches(str, regex) {
 
 let invalid = 0;
 
-for (const file of files) {
-  const filePretty = chalk.bold(
-    path
-      .relative(__dirname, file)
-      .replace(/^public/, "")
-      .replace(/index.html$/, "")
-  );
-  const contents = fs.readFileSync(file, "utf8");
-  if (contents.indexOf("Postgraphile") >= 0) {
-    invalid++;
-    console.error(
-      `${filePretty} mentions 'Postgraphile'; please change to 'PostGraphile' or 'postgraphile' for consistency.`
+const fileLinks = files
+  .map(file => {
+    const filePretty = chalk.bold(
+      path
+        .relative(__dirname, file)
+        .replace(/^public/, "")
+        .replace(/index.html$/, "")
     );
-  }
-  const links = allMatches(contents, /<a[^>]+href="([^"]+)"/g);
-  for (const link of links) {
+    const contents = fs.readFileSync(file, "utf8");
+    if (contents.indexOf("Postgraphile") >= 0) {
+      invalid++;
+      console.error(
+        `${filePretty} mentions 'Postgraphile'; please change to 'PostGraphile' or 'postgraphile' for consistency.`
+      );
+    }
+
+    // WARNING! Cardinal sin below, shield eyes
+    return allMatches(contents, /<a[^>]+href="([^"]+)"/g).map(link => {
+      return {
+        link: entities.decode(link),
+        filePretty,
+      };
+    });
+  })
+  .reduce((flattened, toFlatten) => flattened.concat(toFlatten), []);
```

Comment:
not using Array#flatMap because I assume we want to support node LTS

Reply:
Nah; since it’s the website feel free to bump min deps to Node 12.